### PR TITLE
Converting SP dates to readable excel date format

### LINKED ISSUE #487: Porteføljeoversikt - dates in excel report is not recognized and is difficult to parse
**Describe the bug**
I porteføljeoversikt kan man eksportere data til Excel. Det er vanvittig irriterende at startdato og sluttdato for prosjektene kommer over med feil dato, og på et format som ikke er kompatibelt med Excel. 

**To Reproduce**
1. Navigate to 'Porteføljeoversikt', e.g.

https://universityofbergen.sharepoint.com/sites/Admprosjekter/SitePages/Portef%C3%B8ljeoversikt.aspx#viewId=6

2. Download excel report
3. Open in excel, convert imported data to table (ctrl+t)

Results, dates are not formatted, even when explicitly setting the cell format as "date"

1) Startdato 11.mai 2021 blir i Excel 2021-05-10T22:00:00Z
2) Sluttdato 18.12.2024 blir i Excel 2024-12-17T23:00:00Z

**Expected behavior**
Dates are recognized by excel and parsed correctly. Time of day is not included, as its not needed.

**Additional context**
A workaround is to use a separate column to parse dates using formulas, e.g.

`=DATEVALUE(MID(G2;1;10))+TIMEVALUE(MID(G2;12;8))`

A sample report is attached. Note that the two last columns has been added and is using the above formula as a workaround.

[sample-report.xlsx](https://github.com/Puzzlepart/prosjektportalen365/files/6701762/sample-report.xlsx)

### PR DESCRIPTION
**Pull request by Rundez**
### Your checklist for this pull request

- [X] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your main!
- [X] Make sure you are making a pull request against the **dev** branch (left side). Also you should start *your branch* off *dev*.
- [X] Check the commit's or even all commits' message 
- [X] Check if your code additions will fail linting checks
- [X] Remember: Add PR description to [CHANGELOG](https://github.com/Puzzlepart/prosjektportalen365/blob/dev/CHANGELOG.md) with the ID that matches this PR

### Description

When exporting project information with dates from "Porteføljeoversikt", the dates were not recognized by excel due to the format. 
The solution: Convert the Sharepoint date format to a standard JS date object recognized by excel. 

### How to test

1. Navigate to "Porteføljeoversikt", add start and end date columns, then export to excel. 
2. The dates should now be recognized by excel. 

### Relevant issues (if applicable)
fixes #487 

💔Thank you!

### CODE REVIEW

**Review by olemp — CHANGES_REQUESTED** [review]
Too hardcoded in my opinion. Only works for these two date fields. We should look into a more dynamic solution that works for all date fields, including custom ones.

**Comment by tarjeieo** [comment]
> Too hardcoded in my opinion. Only works for these two date fields. We should look into a more dynamic solution that works for all date fields, including custom ones.

I agree that it's not ideal. How about iterating over properties of the item, converting those ending with *OWSDATE to proper date values?

**Comment by olemp** [comment]
> > Too hardcoded in my opinion. Only works for these two date fields. We should look into a more dynamic solution that works for all date fields, including custom ones.
> 
> I agree that it's not ideal. How about iterating over properties of the item, converting those ending with *OWSDATE to proper date values?

We should have access to the columns array, where we have the column config and can catch if it's a date.

**Comment by Rundez** [comment]
I agree. Added a solution that iterates the columns and checks the field type. This should also convert any custom date values.

**Comment by tarjeieo** [comment]
Any input here @olemp ? Can this be merged?

**Comment by olemp** [comment]
> Any input here @olemp ? Can this be merged?

Still want to have a look through it. Should have time this week - tomorrow or Thu.

**Comment by tarjeieo** [comment]
> > Any input here @olemp ? Can this be merged?
> 
> Still want to have a look through it. Should have time this week - tomorrow or Thu.

Friendly reminder :)

File: SharePointFramework/PortfolioWebParts/src/components/PortfolioOverview/PortfolioOverviewCommands/index.tsx [hunk 1]
```diff
@@ -173,7 +173,17 @@ export class PortfolioOverviewCommands extends Component<
     this.setState({ isExporting: true })
     try {
       const { fltItems, fltColumns, selectedItems } = this.props
+
       const items = isArray(selectedItems) && selectedItems.length > 0 ? selectedItems : fltItems
+
+      fltColumns.map((col) => {
```
Comment (olemp) [inline SharePointFramework/PortfolioWebParts/src/components/PortfolioOverview/PortfolioOverviewCommands/index.tsx:179]: @Rundez You're using `Array.map` without returning anything. Seems like it's `Array.forEach` you want to use here.

**Review by olemp — CHANGES_REQUESTED** [review]
Se comments.

**Comment by olemp** [comment]
See [Array.map documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)

**Comment by danielwiese** [comment]
Tested OK

**Pull request merged**